### ISSUE TITLE
Loki integration with alertmanager requires additional config

### ISSUE THREAD

**Issue by pengwyn** [canonical/loki-k8s-operator#379]
### Bug Description

With the latest versions of the charms, and when loki is provided with some alert rules, it fails to publish these to alertmanager due to an API version issue. 
I found that a config option [1] is required for Loki:

```
ruler:
    enable_alertmanager_v2: true
```

which is not present in the charm configuration.

[1] https://grafana.com/docs/loki/latest/alert/#scheduling-and-best-practices

### To Reproduce

1. Deploy a cos cluster
2. Create a set of loki alert rules and publish them to loki via the `cos-configuration-k8s` charm.

### Environment

Juju 3.4. Model config:

```
App                              Version  Status   Scale  Charm                         Channel        Rev  Address         Exposed  Message
alertmanager                     0.27.0   active       1  alertmanager-k8s              latest/edge    108  10.152.183.247  no       
catalogue                                 active       1  catalogue-k8s                 latest/stable   33  10.152.183.228  no       
cos-configuration-k8s            3.5.0    active       1  cos-configuration-k8s         latest/stable   46  10.152.183.168  no
grafana                          9.2.1    active       1  grafana-k8s                   latest/stable  106  10.152.183.84   no       
loki                             2.9.4    active       1  loki-k8s                      latest/stable  124  10.152.183.17   no       
prometheus                       2.49.1   active       1  prometheus-k8s                latest/stable  171  10.152.183.40   no       
traefik                          2.10.5   active       1  traefik-k8s                   latest/stable  174  10.64.140.43    no       
```

### Relevant log output

```shell
The error reported in the pod logs is:

alertmanager=http://alertmanager-0.alertmanager-endpoints.cos.svc.cluster.local:9093/api/v1/alerts count=1 msg="Error sending alert" err="bad response status 410 Gone"

and when I manually query the alertmanager at that URL I get the response:

{"status":"deprecated","error":"The Alertmanager v1 API was deprecated in version 0.16.0 and is removed as of version 0.28.0 - please use the equivalent route in the v2 API"}
```

**Comment by cbartz** [comment]
+1

**Comment by ca-scribner** [comment]
Closed with #396.  If we're mistaken, please reopen
